app/models/application.rb

```diff
@@ -0,0 +1,26 @@
+class Application < ApplicationRecord
+  has_many :pet_applications
+  has_many :pets, through: :pet_applications
+ 
+  validates_presence_of :name, :street, :city, :state, :zip, :descr, :status
+
+  enum status: {"In progress": 0, "Pending": 1, "Approved": 2, "Rejected": 3}
+
+  def pet_search(name)
+    Pet.where("name = ?", name)
+  end
+
+  def get_pet_app(pet_id) 
+    PetApplication.where(pet_id: pet_id, application_id: self.id).first
+  end
+
+  def all_pets_apps_appr?
+    if self.pet_applications.where('status = 2').count == self.pet_applications.count &&
```

Comment:
I wonder if you could simplify this logic by instead using `where("status != 2")` to find any cases where there are pet_apps that aren't approved. The elsif statement here is not as readable but I understand what you're accomplishing here!